app/services/html_parser.rb

```diff
@@ -1,16 +1,37 @@
-module CodeBlockParser
+class HtmlParser
+  # Each of the instance methods should return self to support chaining of
+  # methods
+  # For example:
+  #  HtmlParser.
+  #    new(html).
+  #    remove_nested_linebreak_in_list.
+  #    prefix_all_images.
+  #    wrap_all_images_in_links.
+  #    html
+
   include InlineSvg::ActionView::Helpers
   include ApplicationHelper
 
-  def remove_nested_linebreak_in_list(html)
-    html_doc = Nokogiri::HTML(html)
+  RAW_TAG_DELIMITERS = ["{", "}", "raw", "endraw", "----"].freeze
+
+  attr_reader :html
```

Comment:
I see this is an attribute reader, is there a specific reason why you opted not to make this an accessor, to you could assign to it? I see down that you assign to `@html` a bunch of times

Reply:
It prevents anyone from reassigning `html` by doing something like:

```ruby
html_parser = HtmlParser.
  new(html).
  some_chained_method.
  html
  
html_parser.html = some_new_html
```

Currently, that re-assignment at the end wouldn't work because it would fail with undefined method `html=`. It would work if we made it an accessor. I'm open to making it an accessor though, I don't expect this would happen often, if at all. Admittedly, I'm probably being more careful than necessary.

@rhymes What do you think, should we change it?

Reply:
Ah right! Make sense, maybe the solution is to split it: attr_reader stays public, attr_writer is moved down as private. That might work

Reply:
Good idea, updated!